src/main/java/it/pagopa/pn/deliverypush/action/refinement/RefinementHandler.java

```diff
@@ -41,7 +41,7 @@ public void handleRefinement(String iun, Integer recIndex) {
 
         //Se la notifica è già stata visualizzata non viene perfezionata per decorrenza termini in quanto è già stata perfezionata per presa visione
         if( !isNotificationViewed ) {
-            addRefinementElement(iun, recIndex, pnDeliveryPushConfigs.getRetentionAttachmentDaysAfterRefinement());
+            addRefinementElement(iun, recIndex, pnDeliveryPushConfigs.getRetentionAttachmentDaysAfterRefinement(),true);
```

Comment:
In questo caso non siamo sicuri al 100% che la notifica vada fatturata, potrebbe essere già stata annullata.

Reply:
NB: il check sul cancelled è fatto "a monte". Però di fatto non inserirebbe nemmeno l'evento di refinement